libs/libc/signal/sig_delset.c

```diff
@@ -79,3 +78,37 @@ int sigdelset(FAR sigset_t *set, int signo)
       return OK;
     }
 }
+
+/****************************************************************************
+ * Name: sigdelset
+ *
+ * Description:
+ *   This function deletes the signal specified by signo from the signal
+ *   set specified by the 'set' argument.
+ *
+ * Input Parameters:
+ *   set - Signal set to delete the signal from
+ *   signo - Signal to delete
+ *
+ * Returned Value:
+ *   0 (OK), or -1 (ERROR) if the signal number is invalid.
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+int sigdelset(FAR sigset_t *set, int signo)
+{
+  int ret;
+
+  /* Let nxseg_delset do all the work. */
+
+  ret = nxsig_delset(set, signo);
+  if (ret < 0)
+    {
+      set_errno(EINVAL);
```

Comment:
In the general case, it should set_errno(-ret); In this case, -EINVAL should be the only value returned by nxsig_delset() so it does not really matter.